server/github.go

```diff
@@ -48,3 +59,30 @@ func (ctx *context) githubAuth(req *http.Request) bool {
 	expectedMAC := mac.Sum(nil)
 	return hmac.Equal([]byte(sign), []byte(fmt.Sprintf("sha1=%x", expectedMAC)))
 }
+
+func (ctx *context) startGithubJob(r *request) (*response, error) {
+	var payload githubPayload
+
+	r.parseBody(&payload)
+
+	var ref string
+	if branchRegexp.MatchString(payload.Ref) {
+		submatch := branchRegexp.FindAllStringSubmatch(payload.Ref, -1)
```

Comment:
Why do you use the `FindAllStringSubmatch` and not simply `FindStringSubmatch` ?